src/Popover/Popover.tsx

```diff
@@ -0,0 +1,235 @@
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    shift,
+    useDismiss,
+    useRole,
+    useInteractions,
+    useMergeRefs,
+    Placement,
+    useId,
+    useFocus,
+} from "@floating-ui/react";
+import { ButtonHTMLAttributes, Dispatch, HTMLProps, ReactNode, SetStateAction, cloneElement, createContext, forwardRef, isValidElement, useContext, useLayoutEffect, useMemo, useState } from "react";
+
+import Styles from "./Popover.module.scss";
+
+interface PopoverOptions {
+    initialOpen?: boolean;
+    placement?: Placement;
+    modal?: boolean;
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+}
```

Comment:
Better to put these interfaces in a separate file, me be under i.e. `./src/Popover/Popover.props.ts`.
Also a better practice is to put `I` before the interface name. like `IPopoverOptions`.